### PR TITLE
[Feature][e2e] Add Flink E2E

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
Container testing might be better if we create a new CI for it.

File: pom.xml [hunk 1]
```diff
@@ -80,6 +80,7 @@
         <module>seatunnel-connectors</module>
         <module>seatunnel-dist</module>
         <module>seatunnel-examples</module>
+        <module>seatunnel-e2e</module>
```
Comment (CalvinKirs) [inline pom.xml:83]: Maybe it would be better to create a separate project for him, I mean, he doesn't belong to a module of the main project.

Comment (ruanwenjun) [reply pom.xml:83]: @CalvinKirs Thanks for your review, I got you. Do we also need to split e2e with integration test? In some projects, they only have IT and some have both e2e, IT. If we also need IT, I think put IT in the `seatunnel` project is OK. Maybe this PR is more like to only add IT.

Agree with use of another CI for e2e.

Comment (CalvinKirs) [reply pom.xml:83]: That's better. For most people, if the local UT can't run quickly, then they may lose the ability to continue running the UT to verify the overall correctness of this change. We should keep the UT lightweight. So I suggest starting a separate e2e test.

Comment (ruanwenjun) [reply pom.xml:83]: I add the switch to control when to run IT. Now, the IT case will only be executed by set `-DskipIT=false`.

Comment (wuchunfu) [reply pom.xml:83]: > That's better. For most people, if the local UT can't run quickly, then they may lose the ability to continue running the UT to verify the overall correctness of this change. We should keep the UT lightweight. So I suggest starting a separate e2e test.

+1

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**